src/internal/_xtake.js

```diff
@@ -14,8 +14,16 @@ module.exports = (function() {
         return this.xf.result(result);
     };
     XTake.prototype.step = function(result, input) {
-        this.n -= 1;
-        return this.n >= 0 ? this.xf.step(result, input) : _reduced(result);
+        switch (this.n) {
+            case 0:
+                return _reduced(result);
```

Comment:
It should not be.  Calling `step` after a `reduced` would be a bug in the [transducible process](http://clojure.org/transducers#toc11)